src/lib.rs

```diff
@@ -3,15 +3,34 @@ extern crate libc;
 #[cfg(target_os = "openbsd")]
 mod openbsd;
 
+use std::ffi::{CString, NulError};
+
 #[derive(Debug, PartialEq)]
 pub enum Error {
     NotSupported,
+    Path(NulError),
+    Permissions(NulError),
```

Comment:
Couldn't this be replaced with a single `InvalidArgument` case or do you have a use case in mind where this distinction would be useful?

Reply:
not really, but in general, i believe error types should be as detailed as possible. callers can easily ignore details they don’t care about, and it won’t cost us much to leave this in.